conformance/tests/httproute-request-mirror.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteRequestMirror)
+}
+
+var HTTPRouteRequestMirror = suite.ConformanceTest{
+	ShortName:   "HTTPRouteRequestMirror",
+	Description: "An HTTPRoute with request mirror filter",
+	Manifests:   []string{"tests/httproute-request-mirror.yaml"},
+	// Features:    []suite.SupportedFeature{suite.SupportHTTPRouteHostRewrite},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		ns := "gateway-conformance-infra"
+		routeNN := types.NamespacedName{Name: "request-mirror", Namespace: ns}
+		gwNN := types.NamespacedName{Name: "same-namespace", Namespace: ns}
+		gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+
+		testCases := []http.ExpectedResponse{
+			{
+				Request: http.Request{
+					Path: "/mirror",
+				},
+				ExpectedRequest: &http.ExpectedRequest{
+					Request: http.Request{
+						Path: "/mirror",
+					},
+				},
+				Backend:    "infra-backend-v1",
+				MirroredTo: "infra-backend-v2",
+				Namespace:  ns,
+			},
+		}
+		for i := range testCases {
+			// Declare tc here to avoid loop variable
+			// reuse issues across parallel tests.
+			tc := testCases[i]
+			t.Run(tc.GetTestCaseName(i), func(t *testing.T) {
+				t.Parallel()
+				http.MakeRequestAndExpectEventuallyConsistentResponse(t, suite.RoundTripper, suite.TimeoutConfig, gwAddr, tc)
+				if tc.MirroredTo != "" {
```

Comment:
is this `if tc.MirroredTo != "" {` required, we want to ensure we are validating mirrored backends in this test/file(httproute-request-mirror.go)

Reply:
right, fail if `tc.MirroredTo = ""` sounds good ? or you might have a better idea?

Reply:
I'd recommend removing the check altogether, and directly using the val

Reply:
@arkodg  I feel like using the val when its nil will just send unwanted requests to the api server asking for logs from `""` pod name

Reply:
that shouldnt be happening, you could move the check into `ExpectMirroredRequest` and then fail the test if `MirroredTo` is unset